orttraining/orttraining/training_ops/cuda/math/scale_impl.h

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+namespace onnxruntime {
+namespace cuda {
+
+template <typename ScaleT>
+struct GetScaleValueImpl {
+  void operator()(const Tensor* scale, float& scale_value) const {
+    ORT_ENFORCE(scale->Shape().Size() == 1, "Scale input should have a single value.");
+    scale_value = static_cast<float>(*(scale->template Data<ScaleT>()));
+    ORT_ENFORCE(scale_value != 0.0f, "Scale value must not be 0.");
+  }
+};
+
+template <typename T>
+void Impl_Scale(
+    cudaStream_t stream,
```

Comment:
>cudaStream_t [](http://example.com/codeflow?start=4&length=12)

general comment: is it possible to include the header that defines cudaStream_t in these _impl headers too?

Reply:
I hesitate doing it, getting into the headers nightmare.